### ISSUE TITLE
ERROR The OpenSearch REST endpoint has NOT become accessible in the expected time; exiting

### ISSUE THREAD

**Issue by dougmanoel** [sassoftware/viya4-monitoring-kubernetes#364]
Hi,
This error is happening when deploying:

(⎈|jre-prod:jreprod)➜  logging git:(master) ✗ bash -xv /home/domora/JRE/JRE-60/monitoring_logging_issue_2/viya4-monitoring-kubernetes/logging/bin/deploy_logging.sh 2>~/tmp.tmp
INFO User directory: /home/domora/JRE/JRE-60/monitoring_logging_issue_2/cis-viya4-monitoring-kubernetes/tls
INFO Helm client version: 3.8.2
INFO Kubernetes client version: v1.21.1
INFO Kubernetes server version: v1.22.6
INFO Loading user environment file: /home/domora/JRE/JRE-60/monitoring_logging_issue_2/cis-viya4-monitoring-kubernetes/tls/logging/user.env

Deploying logging components to the [logging] namespace [Tue Jun 14 20:55:55 -03 2022]

INFO Deploying Event Router ...
serviceaccount/eventrouter unchanged
clusterrole.rbac.authorization.k8s.io/eventrouter unchanged
clusterrolebinding.rbac.authorization.k8s.io/eventrouter unchanged
configmap/eventrouter-cm unchanged
deployment.apps/eventrouter unchanged
INFO Event Router has been deployed

configmap "run-securityadmin.sh" deleted
configmap/run-securityadmin.sh created
configmap/run-securityadmin.sh labeled
"opensearch" already exists with the same configuration, skipping
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "opensearch" chart repository
...Successfully got an update from the "ingress-nginx" chart repository
...Successfully got an update from the "fluent" chart repository
...Successfully got an update from the "prometheus-community" chart repository
Update Complete. ⎈Happy Helming!⎈
INFO Deploying OpenSearch
Release "opensearch" has been upgraded. Happy Helming!
NAME: opensearch
LAST DEPLOYED: Tue Jun 14 20:56:57 2022
NAMESPACE: logging
STATUS: deployed
REVISION: 5
TEST SUITE: None
NOTES:
Watch all cluster members come up.
  $ kubectl get pods --namespace=logging -l app.kubernetes.io/component=v4m-search -w
INFO Waiting on OpenSearch pods to be Ready
pod/v4m-search-0 condition met
INFO OpenSearch has been deployed

INFO Loading Content into OpenSearch
ERROR The OpenSearch REST endpoint has NOT become accessible in the expected time; exiting.
ERROR Review the OpenSearch pod's events and log to identify the issue and resolve it before trying again.
ERROR Exiting script [deploy_logging.sh] due to an error executing the command [logging/bin/deploy_opensearch_content.sh].
(⎈|jre-prod:jreprod)➜  logging git:(master) ✗

Already saw that there is a 503 error coming from that endpoint from debugging the script deploy_opensearch_content.sh:
+ for pause in 30 30 30 30 30 30 60
++ curl -s -o /dev/null -w '%{http_code}' -XGET https://localhost:41829 --user admin:<THE PASSWORD HERE> --insecure
+ response=503

Checked pod logs and says:
[2022-06-15T00:10:10,469][ERROR][o.o.s.a.BackendRegistry  ] [v4m-search-0] Not yet initialized (you may need to run securityadmin)

Which kind of matches somehow with comments in that loop to verify the endpoint:
   "returns 503 (and outputs "Open Distro Security not initialized.") when ODFE isn't ready yet"
   "TO DO: check for 503 specifically?"

What to do resolve that?

**Comment by gsmith-sas** [comment]
As @dougmanoel indicated, we were able to resolve the original problem.  That problem was caused by an incorrect host name in the $USER_DIR/logging/user-values-opensearch.yaml file.  

I'm following up via e-mail with @dougmanoel on the new unscheduled Fluent Bit pod issue he mentioned...which is a separate issue.

I am going to close this issue (since the original problem is resolved).
